exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/DrillFunctionRegistry.java

```diff
@@ -92,23 +118,106 @@ public DrillFunctionRegistry(ScanResult classpathScan) {
   }
 
   public int size(){
-    return methods.size();
+    return registeredFunctions.size();
   }
 
   /** Returns functions with given name. Function name is case insensitive. */
   public List<DrillFuncHolder> getMethods(String name) {
-    return this.methods.get(name.toLowerCase());
+    return this.registeredFunctions.get(name.toLowerCase());
   }
 
   public void register(DrillOperatorTable operatorTable) {
```

Comment:
Also, is there a way without registering the two variants of the same UDF?

Implementations of PluggableFunctionRegistry need to know call operatorTable#addInference and operatorTable#addDefault methods.